vochain/process.go

```diff
@@ -176,6 +177,45 @@ func (v *State) SetProcessStatus(pid []byte, newstatus models.ProcessStatus, com
 	return nil
 }
 
+// SetProcessResults sets the results for a given process. Only if the process status
+// allows and the format of the results allows to do so
+func (v *State) SetProcessResults(pid []byte, result *models.ProcessResult, commit bool) error {
+	process, err := v.Process(pid, false)
+	if err != nil {
+		return err
+	}
+	// Check if the state transition is valid
+	switch process.Status {
+	case models.ProcessStatus_ENDED:
+		// process must be ended for setting the results
+	case models.ProcessStatus_RESULTS:
+		return fmt.Errorf("results already added")
```

Comment:
Check if added results are the same as ours, else return an error

Reply:
Add a TO-DO: allow result confirm with another Tx (in order to have N/M oracle signatures)